marketplace-client/src/components/pages/AddProduct.js

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+import axios from 'axios';
+
+var loadjs = require('loadjs');
+var email = "";
+
+class AddProduct extends React.Component {
+    componentDidMount() {
+        loadjs('main.js');
+        if (localStorage.getItem('token') !== null) {
+            axios({
+                method: 'POST',
+                url: 'users/authToken',
+                headers: {
+                    "Content-Type": "application/json",
+                    'token': localStorage.getItem('token')
+                }
+            }).then((response) => {
+                email = response.data.email;
+            }).catch((error) => {
+                console.log(error);
+            });
+        } else {
+            window.location.href = "./";
+        }
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            name: '',
+            description: '',
+            categoryid: '',
+            price: '',
+            city: '',
+            location: '',
+            address: '',
+            saletype: '',
+            errorMessage: '',
+            image: null
+        }
+    }
+
+
+    buildStateOptions() {
+        var arr = [];
+
+        const states = ['Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California', 'Colorado',
+            'Connecticut', 'Delaware', 'Florida', 'Georgia', 'Hawaii', 'Idaho',
+            'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine',
+            'Maryland', 'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana',
+            'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico', 'New York', 'North Carolina',
+            'North Dakota', 'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania', 'Rhode Island', 'South Carolina',
+            'South Dakota', 'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington',
+            'West Virginia', 'Wisconsin', 'Wyoming'];
+
+        for (let i = 0; i <= states.length - 1; i++) {
+            arr.push(<option key={i} value={states[i]}>{states[i]}</option>);
+        }
+
+        return arr;
+    }
+
+    buildSaleTypeOptions() {
+        var arr = [];
+
+        const saleTypes = ['Community', 'Services', 'For Sale', 'For Rent'];
+
+        for (let i = 0; i <= saleTypes.length - 1; i++) {
+            arr.push(<option key={i} value={saleTypes[i]}>{saleTypes[i]}</option>);
+        }
+
+        return arr;
+    }
+
+    buildCategoryTypeOptions() {
+        var arr = [];
+
+        const categories = ['Artists', 'Apartments', 'Homes',
+            'Offices / Commercal Space', 'Cleaning Service', 'Furniture', 'Cooking', 'Transportation'];
+
```

Comment:
Typo in Commercial is preventing any post of category offices / commercial space from getting fetched, will commit that change